src/main/java/org/folio/controller/AllowedServicePointsController.java

```diff
@@ -25,45 +26,61 @@
 @AllArgsConstructor
 public class AllowedServicePointsController implements AllowedServicePointsApi {
 
-  private final AllowedServicePointsService allowedServicePointsService;
+  private final ItemLevelServicePointServiceImpl itemLevelSpService;
+  private final TitleLevelServicePointServiceImpl titleLevelSpService;
```

Comment:
Please don't inject concrete implementations. 

The cleanest approach here would be to create a AllowedServicePointFactory class, inject it into the controller instead of the two service implementations, and delegate choosing the implementation at runtime to this factory.

Second best option (IMHO) is to use your approach with factory method, but inject both services as objects of `AllowedServicePointsService`. For this you can name your implementations, for example as: 
```
@Service("allowedServicePointsForItemLevelRequestService")
and
@Service("allowedServicePointsForTitleLevelRequestService")
```
and then inject both services into the controller in at least two different ways:
- by using `@Qualifier` annotation on both injected instances: `@Qualifier("allowedServicePointsForItemLevelRequestService")`
- or by naming you injected instances according to bean names: `private final AllowedServicePointsService allowedServicePointsForItemLevelRequestService`